### DISCUSSION TITLE
A way to disable the `ESC` key behavior of quiting a sketch

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#548] [Ideas]
In Processing Python mode I have seen some examples with code like this:

```python
def keyPressed():
    if key == ESC:
        # do whatever you want ESC to trigger
        this.key = ' '  # this would allow the sketch to continue running
```

But that doesn't work on py5 as the current sketch object, obtained with `get_current_sketch()`, won't allow fiddling with the `key` attribute. Any suggestions?

**Comment by hx2A** [comment]
Interesting. I looked at the Processing source code and I see the comments specifically mention setting `key` to something else:

```java
...
    // if someone else wants to intercept the key, they should
    // set key to zero (or something besides the ESC).
    if (event.getAction() == KeyEvent.PRESS) {
      //if (key == java.awt.event.KeyEvent.VK_ESCAPE) {
      if (key == ESC) {
        exit();
      }
...
```

I agree that what you are doing should be possible. I can't think of a way to do this without making the `key` property settable. This is certainly possible. Normally the properties are read only. Perhaps in Processing you can fiddle with variables like `width` and `height` but I don't think allowing this is a good idea. Allowing `key` to be modified makes sense though.

The alternative is to have some special `set_key()` method, but that seems weird and awkward to me, and totally different from how Processing works.

I'll open an issue for this so I don't forget about it. It won't be in tomorrow's release but it will be in the next release after that one.

**Reply by villares** [reply]
The issue about this was https://github.com/py5coding/py5generator/issues/549, a fix to be released soon!
